hooks/_common.sh

```diff
@@ -219,10 +219,21 @@ function common::per_dir_hook {
 
   # Lookup hook-config for modifiers that impact common behavior
   local change_dir_in_unique_part=false
+  # Limit the number of parallel processes to the number of CPU cores -1
+  # `nproc` - linux, `sysctl -n hw.ncpu` - macOS, `echo 1` - fallback
+  local CPU
+  CPU=$(nproc || sysctl -n hw.ncpu || echo 1)
```

Comment:
I seem to recall there was something why declaration need to be split off from definition, but I can't remember what exactly, and hence:
```suggestion
  local CPU=$(nproc || sysctl -n hw.ncpu || echo 1)
```
ps: works for me locally =)

Reply:
![image](https://github.com/antonbabenko/pre-commit-terraform/assets/11096782/f11b9a91-a20a-4e44-9314-22a394af4a05)

Reply:
Ah, this.
The funny bit is that this shell check is targeted against masking return values while the logic of this code bit is to mask errors and we don't care of its return status.
By the way we may want to redirect stderr to `/dev/null` for each of first two commands to stop vomiting console with pointless output:
```shell
> cat ~/tmp/zzz.sh
#!/usr/bin/env bash

zzz() {
        local TEST=$(should_fail || sysctl -n hw.ncpu || echo 1) 2>/dev/null
        echo $TEST
}

zzz

> ~/tmp/zzz.sh

/home/giermulnik/tmp/zzz.sh: line 17: should_fail: command not found
sysctl: cannot stat /proc/sys/hw/ncpu: No such file or directory
1
```